sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -94,6 +94,21 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("create a permanent view on a temp view used inside an expression") {
+    withTable("base_table") {
+      withTempView("temp_view") {
+        sql("CREATE TABLE base_table (id int) USING parquet")
+        sql("CREATE TEMPORARY VIEW temp_view AS SELECT * FROM base_table")
+        val e = intercept[AnalysisException] {
+          sql("CREATE VIEW per_view AS SELECT * FROM base_table " +
+            "WHERE EXISTS (SELECT * FROM temp_view)")
```

Comment:
Can you add more exhaustive test patterns for subqueries? e.g., nested subqueries and subqueries in SELECT stmts like `select c, (select count(*) from temp) from base`?

Reply:
I added few more as suggested. Please let me know if more are needed.